lib/presentation/book/view/home_page_view.dart

```diff
@@ -9,8 +16,149 @@ class HomePageView extends StatefulWidget {
 }
 
 class _HomeBookPageView extends State<HomePageView> {
+  late final _controller;
+  late List<Book> bookList = [];
+
+  @override
+  void initState() {
+    context.read<HomePageBloc>().add(DownloadBooks());
+
+    _controller = PageController(viewportFraction: 0.9, initialPage: 0);
+
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
   @override
   Widget build(BuildContext context) {
-    return Scaffold();
+    return BlocConsumer<HomePageBloc, HomePageState>(
+      listener: (context, state) {
+        if (state is LoadingState) {
+          DialogUtils.showLoadingScreen(context);
+        } else if (state is LoadedState) {
+          Navigator.pop(context);
+        } else if (state is SuccessfulBookAdded) {
+          context.read<HomePageBloc>().add(DownloadBooks());
+        } else if (state is BooksDownloadedState) {
+          bookList.addAll(state.bookList);
```

Comment:
this is a bug. You will always add books when this state happens. You need to clear bookList before you add. Or just bookList = List.from(state.bookList)